tests/io/delta_lake/conftest.py

```diff
@@ -170,28 +179,104 @@ def s3_path(
+### Azure-specific fixtures ###
+###############################
+
+
+@pytest.fixture(scope="session")
+def az_credentials() -> dict[str, str]:
+    return {
+        # These are the well-known values
+        # https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio#well-known-storage-account-and-key
+        "ACCOUNT_NAME": "devstoreaccount1",
+        "KEY": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==",
+    }
+
+
+@pytest.fixture(scope="session")
+def az_server_ip() -> str:
+    return "0.0.0.0"
+
+
+@pytest.fixture(scope="session")
+def az_server_port() -> int:
+    return 10000
+
+
+@pytest.fixture(scope="session")
+def az_server(az_server_ip: str, az_server_port: int) -> Iterator[str]:
+    az_server_url = f"http://{az_server_ip}:{az_server_port}"
+    client = docker.from_env()
+    azurite = client.containers.run(
+        "mcr.microsoft.com/azure-storage/azurite",
+        f"azurite-blob --loose --blobHost {az_server_ip}",
+        detach=True,
+        ports={str(az_server_port): str(az_server_port)},
+    )
+    for _ in range(100):
+        if azurite.status == "running":
+            break
+        elif azurite.status == "exited":
+            output = azurite.logs()
+            print(f"Azurite container exited without executing tests: {output}")
+            pytest.fail(f"Cannot start mock Azure Blob Storage server")
+        time.sleep(0.1)
+        azurite.reload()
+    try:
+        yield az_server_url
+    finally:
+        azurite.stop()
+
+
+@pytest.fixture(scope="function")
+def az_path(data_dir: str, az_server: str, az_credentials: dict[str, str]) -> Iterator[tuple[str, daft.io.IOConfig]]:
+    path = data_dir.strip("/").replace("_", "-")
+    account_name = az_credentials["ACCOUNT_NAME"]
+    key = az_credentials["KEY"]
+    endpoint_url = f"{az_server}/{account_name}"
+    conn_str = f"DefaultEndpointsProtocol=http;AccountName={account_name};AccountKey={key};BlobEndpoint={endpoint_url};"
+
+    io_config = daft.io.IOConfig(
+        azure=daft.io.AzureConfig(
+            storage_account=az_credentials["ACCOUNT_NAME"],
+            access_key=az_credentials["KEY"],
+            endpoint_url=endpoint_url,
+            use_ssl=False,
+        )
+    )
+    bbs = BlobServiceClient.from_connection_string(conn_str)
+    container = path.split("/")[0]
+    bbs.create_container(container)
+    try:
+        yield "az://" + path, io_config
+    finally:
+        bbs.delete_container(container)
 
 
 @pytest.fixture(scope="function")
-def local_path(tmp_path: pathlib.Path, data_dir: str) -> (str, None):
+def local_path(tmp_path: pathlib.Path, data_dir: str) -> tuple[str, None]:
     path = os.path.join(tmp_path, data_dir)
     os.mkdir(path)
-    yield path, None
+    return path, None
 
 
 @pytest.fixture(
     scope="function",
     params=[
         lazy_fixture("local_path"),
         lazy_fixture("s3_path"),
+        lazy_fixture("az_path"),
```

Comment:
I still do think it's a good idea to run the `az_path` fixture as an integration test instead

If a new contributor comes without Docker, they should still be able to run unit tests (without some tests implicitly skipping).

To run integration tests, they should install Docker!

Reply:
Docker is already included as a dev dependency, so if they `pip install -r requirements-dev.txt` (which I think we assume is a testing prerequisite), it will work automatically. https://github.com/Eventual-Inc/Daft/blob/9ed9b3f4dc42965310080ae682eb9f7337b22fa7/requirements-dev.txt#L5

But looking at the uses, it looks like it's only used for integration tests... I'll look into downgrading this to an integration test.

Reply:
I believe this is docker bindings and not docker itself!

Reply:
Yup the `docker` Python dependency is just a Python API client to talk to Docker. We can't control whether or not Docker itself is installed on the system from requirements-dev.txt unfortunately 😬